plugins/rich-editor/src/scripts/components/toolbars/ParagraphDropDown.tsx

```diff
@@ -0,0 +1,139 @@
+/*
+ * @author Stéphane LaFlèche <stephane.l@vanillaforums.com>
+ * @copyright 2009-2018 Vanilla Forums Inc.
+ * @license GPL-2.0-only
+ */
+
+import React from "react";
+import Quill from "quill/core";
+import HeadingBlot from "quill/formats/header";
+import { withEditor, IWithEditorProps } from "@rich-editor/components/context";
+import Formatter from "@rich-editor/quill/Formatter";
+import ParagraphToolbarMenuItems from "@rich-editor/components/toolbars/pieces/ParagraphToolbarMenuItems";
+import CodeBlockBlot from "@rich-editor/quill/blots/blocks/CodeBlockBlot";
+import BlockquoteLineBlot from "@rich-editor/quill/blots/blocks/BlockquoteBlot";
+import SpoilerLineBlot from "@rich-editor/quill/blots/blocks/SpoilerBlot";
+import MenuItems from "@rich-editor/components/toolbars/pieces/MenuItems";
+import FocusWatcher from "@library/FocusWatcher";
+import DropDown from "@library/components/dropdown/DropDown";
+import { ButtonBaseClass } from "@library/components/forms/Button";
+import { heading2, heading3, blockquote, codeBlock, spoiler } from "@library/components/icons/editorIcons";
+import * as editorIcons from "@library/components/icons/editorIcons";
+
+interface IProps extends IWithEditorProps {
+    disabled?: boolean;
+    renderAbove?: boolean;
+    renderLeft?: boolean;
+}
+
+interface IState {
+    hasFocus: boolean;
+}
+
+/**
+ * Implemented ParagraphDropDown component, this is for mobile
+ */
+export class ParagraphDropDown extends React.PureComponent<IProps, IState> {
+    private quill: Quill;
+    private ID: string;
+    private componentID: string;
```

Comment:
`componentID`, `menuID`, and `buttonID` are all set in the constructor, but are not not used anywhere in the component.